### PR TITLE
Add custom eslint rule to avoid new

### PR DESCRIPTION
**Pull request by orangejenny**
@millerdev / @emord what do you think?
also @czue

### CODE REVIEW

**Comment by czue** [comment]
don't feel i have enough of an opinion to be useful on this

File: package.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
   "repository": "https://github.com/dimagi/commcare-hq",
   "description": "CommcareHQ Javascript tests",
   "dependencies": {
+    "eslint-plugin-eslint-dimagi": "file:eslint",
```
Comment (emord) [inline package.json]: A lot of these should probably be `devDependencies` (probably everything except for `js-xpath` and `uglifyjs`

**Review by emord — COMMENTED** [review]
No comment on whether this is the correct thing to do code wise. It's been too long since I've written a lot of JS for me to have an opinon on that.

I think you may need to make this it's own library to get it to work with stickler, which may also be helpful to have this on other projects.

**Review by millerdev — COMMENTED** [review]
It's cool that this took so little code to implement. Looks like getting ~tests to pass on travis~ it working with stickler CI still needs some work.

File: eslint/package.json [hunk 2]
```diff
@@ -15,6 +15,9 @@
   "dependencies": {
     "requireindex": "~1.1.0"
   },
+  "peerDependencies": {
+    "eslint": ">=0.8.0"
+  }
```
Comment (millerdev) [inline eslint/package.json]: missing comma, not sure if that caused the subsequent failure

File: package.json [hunk 3]
```diff
@@ -13,4 +13,7 @@
     "mocha": "^2.3.3",
     "uglifyjs": "^2.4.10"
   }
```
Comment (millerdev) [inline package.json]: missing comma

File: corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ hqDefine("data_dictionary/js/data_dictionary", [
     hqMain,
     googleAnalytics
 ) {
+    var thing = new Thing();
```
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js]: Looks like it worked! Woohoo!!

[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js]: Yes! Mark added custom plugin support to stickler the other day.

**Comment by orangejenny** [comment]
The lint errors within the plugin are because it's both ES6 and a node environment. I'm inclined to ignore them - I'd like to keep the plugin within this repo because I'm expecting its rules to be mostly HQ-specific.

File: eslint/docs/rules/no-unblessed-new.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Disallow new expect for specified objects. (no-unblessed-new)
```
Comment (millerdev) [inline eslint/docs/rules/no-unblessed-new.md]: s/expect/except/

**Review by millerdev — APPROVED** [review]

**Pull request merged**